### PR TITLE
K8s Test Cleanup and Service PTR

### PR DESCRIPTION
**Pull request by johnbelamaric**
Change the CI setup for K8s to be simpler. Now it just creates a
set of objects via a yaml file, making it very easy to modify
the tests.

Implement PTR for services.

### CODE REVIEW

File: middleware/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -53,6 +54,26 @@ func (k *Kubernetes) Services(state request.Request, exact bool, opt middleware.
 	return s, nil, e // Haven't implemented debug queries yet.
 }
 
+// PrimaryZone will return the first non-reverse zone being handled by this middleware
+func (k *Kubernetes) PrimaryZone() (string) {
+	if len(k.primaryZone) > 0 {
```
Comment (miekg) [inline middleware/kubernetes/kubernetes.go]: think `!=""` is more idomatic.

File: middleware/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ import (
 type Kubernetes struct {
 	Next          middleware.Handler
 	Zones         []string
+	primaryZone   string
```
Comment (miekg) [inline middleware/kubernetes/kubernetes.go]: primaryZone is only an index into Zones. Doesn't need to be a string, I guess.

File: middleware/kubernetes/kubernetes.go [hunk 3]
```diff
@@ -286,7 +307,8 @@ func (k *Kubernetes) getServiceRecordForIP(ip, name string) []msg.Service {
 	}
 	for _, service := range svcList {
 		if service.Spec.ClusterIP == ip {
-			return []msg.Service{{Host: ip}}
+			name := k.NameTemplate.RecordNameFromNameValues(nametemplate.NameValues{TypeName: "svc", ServiceName: service.ObjectMeta.Name, Namespace: service.ObjectMeta.Namespace, Zone: k.PrimaryZone()})
+			return []msg.Service{msg.Service{Host: name}}
```
Comment (miekg) [inline middleware/kubernetes/kubernetes.go:296]: does this  work correct when `PrimaryZone` returns the empty string?

Comment (johnbelamaric) [reply middleware/kubernetes/kubernetes.go:296]: In latest version you can't have an empty primary zone because setup will fail.

**Comment by miekg** [comment]
Hmmm, travis now errors with

```
$ $KUBECTL create -f /travis/kubernetes/dns-test.yaml
namespace "test-1" created
namespace "test-2" created
deployment "de-1-a" created
deployment "de-1-b" created
deployment "de-c" created
deployment "de-c" created
Error from server: error when creating "/travis/kubernetes/dns-test.yaml": Service "svc-1-a" is invalid: spec.clusterIP: Invalid value: "10.3.0.100": provided IP is not in the valid range
```

**Comment by johnbelamaric** [comment]
Yep, will fix today.

**Pull request merged**